lab_2_retrieval_w_bm25/main.py

```diff
@@ -62,6 +93,20 @@ def calculate_tf(vocab: list[str], document_tokens: list[str]) -> dict[str, floa
 
     In case of corrupt input arguments, None is returned.
     """
+    if not ((isinstance(vocab, list) and vocab and all(isinstance(word_1, str) for word_1 in vocab)) and
+       (isinstance(document_tokens, list) and document_tokens and
+       all(isinstance(word_2, str) for word_2 in document_tokens))):
+        return None
+
+    all_words = set(vocab).union(document_tokens)
+    tf_dict = {}
+    for t_word in all_words:
+        if t_word in document_tokens:
+            tf_dict[t_word] = document_tokens.count(t_word) / len(document_tokens)
```

Comment:
do it always, but set 0 only if not exists

Reply:
done